### PR TITLE
Unlimited Variables

### PR DESCRIPTION
**Pull request by secsome**
In vanilla YR, local variables are limited up to 100, now you can use up to INT_MAX waypoints.

### CODE REVIEW

**Comment by Thrifinesma** [comment]
Unlimited Global Variables when :)

File: src/Ext/Scenario/Body.h [hunk 1]
```diff
@@ -17,9 +17,13 @@ class ScenarioExt
...
 	public:
 		std::map<int, CellStruct> Waypoints;
+		std::map<int, Variable> LocalVariables;
+		std::map<int, Variable> GlobalVariables;
 
 		ExtData(ScenarioClass* OwnerObject) : Extension<ScenarioClass>(OwnerObject),
-			Waypoints {}
+			Waypoints {},
+			LocalVariables {},
+			GlobalVariables {}
```
Comment (Metadorius) [inline src/Ext/Scenario/Body.h:26]:
```suggestion
			Waypoints { },
			LocalVariables { },
			GlobalVariables { }
```

**Review by Metadorius — CHANGES_REQUESTED** [review]
Overall the code is good but the code for globals and locals is pretty much duplicated. I suggest to remove duplication and merge if there are no any more issues. Also needs docs/readme entries.

**Comment by SMxReaver** [comment]
I do not know the best way to test Save/Load since I am very unfamiliar with how locals function in-game, but I can confirm the game successfully fired Local Variable triggers that were indexed higher than 100. Attached is my test map, I use Local 103 to reveal the map and Local 90 to display "Boris" in the top left corner.
[variabletestmap.zip](https://github.com/Phobos-developers/Phobos/files/6995297/variabletestmap.zip)

**Review by Metadorius — APPROVED** [review]

**Pull request merged**